doc/release/release_dev.rst

```diff
@@ -36,6 +36,8 @@ X contributors. Highlights include:
 Improvements
 ------------
 
+- [`#5216 <https://github.com/networkx/networkx/pull/5216>`_]
+  Make ``omega()`` closer to the published algorithm. The value changes slightly.
```

Comment:
I think we should also mention that the default value for the `niter` parameter is changing from 1->5 in the case of `lattice_reference` and from 100->5 in the case of `omega`